CMakeLists.txt

```diff
@@ -49,36 +49,39 @@ else (NOT BUILD_STATIC)
   set (BUILD_SHARED OFF)
 endif (NOT BUILD_STATIC)
 
-## Retrieve svn revision (at configure time)
-#  Not using Subversion_WC_INFO, as it would have to connect to the repo
-find_program (SVN_EXECUTABLE svn DOC "subversion command line client")
-
-macro (Subversion_GET_REVISION dir variable)
-  execute_process (COMMAND ${SVN_EXECUTABLE} info ${CMAKE_SOURCE_DIR}/${dir}
-    OUTPUT_VARIABLE ${variable}
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-  string (REGEX REPLACE "^(.*\n)?Revision: ([^\n]+).*"
-    "\\2" ${variable} "${${variable}}")
-endmacro (Subversion_GET_REVISION)
+## Retrieve git revision (at configure time)
+find_program (GIT_EXECUTABLE git DOC "git command line client")
+
+macro (Git_GET_REVISION dir variable)
+  execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --abbrev-ref HEAD
+                  WORKING_DIRECTORY ${dir}
+                  OUTPUT_VARIABLE GIT_BRANCH
+                  OUTPUT_STRIP_TRAILING_WHITESPACE)
+  execute_process(COMMAND ${GIT_EXECUTABLE} log -1 --format=%h
+                  WORKING_DIRECTORY ${dir}
+                  OUTPUT_VARIABLE GIT_COMMIT_HASH
+                  OUTPUT_STRIP_TRAILING_WHITESPACE)
+  set (${variable} "${GIT_COMMIT_HASH}-${GIT_BRANCH}")
+endmacro (Git_GET_REVISION)
 
 if (NOT CMAKE_BUILD_TYPE MATCHES "Release")
-  if (EXISTS "${CMAKE_SOURCE_DIR}/.svn/" OR EXISTS "${CMAKE_SOURCE_DIR}/../.svn/")
-    if (SVN_EXECUTABLE)
-      Subversion_GET_REVISION(. ProjectRevision)
-      set (SVN_REVISION "~svn${ProjectRevision}")
-    else (SVN_EXECUTABLE)
-      set (SVN_REVISION "~svn")
-    endif (SVN_EXECUTABLE)
-  endif (EXISTS "${CMAKE_SOURCE_DIR}/.svn/" OR EXISTS "${CMAKE_SOURCE_DIR}/../.svn/")
+  if (EXISTS "${CMAKE_SOURCE_DIR}/.git/" OR EXISTS "${CMAKE_SOURCE_DIR}/../.git/")
+    if (GIT_EXECUTABLE)
+      Git_GET_REVISION(. ProjectRevision)
+      set (GIT_REVISION "~git${ProjectRevision}")
```

Comment:
I think a '-' (or other separator) between 'git' and the actual revision would improve readability, otherwise `CPACK_PACKAGE_VERSION` ends up as `1.0+beta1~gita15caa79-gitrevision`.

Reply:
Done